tests/0105-transactions_mock.c

```diff
@@ -564,7 +569,120 @@ static void do_test_txn_fenced_reinit(rd_kafka_resp_err_t error_code) {
         TEST_SAY("Fatal error: %s: %s\n", rd_kafka_err2name(fatal_err), errstr);
 
         /* All done */
+        rd_kafka_destroy(rk);
+
+        allowed_error = RD_KAFKA_RESP_ERR_NO_ERROR;
+
+        SUB_TEST_PASS();
+}
+
+
+/**
+ * @brief KIP-360: Test that fatal idempotence errors triggers abortable
+ *        transaction errors, but let the broker-side abort of the
+ *        transaction fail with a fencing error.
+ *        Should raise a fatal error.
+ *
+ * @param error_code Which error code EndTxn should fail with.
+ *                   Either RD_KAFKA_RESP_ERR_INVALID_PRODUCER_EPOCH (older)
+ *                   or RD_KAFKA_RESP_ERR_PRODUCER_FENCED (newer).
+ */
+static void do_test_txn_fenced_abort(rd_kafka_resp_err_t error_code) {
```

Comment:
Good test :+1:

Does the unfixed txnmgr implementation fail this test?

Reply:
yes, it fails with
`### Test "0105_transactions_mock (do_test_txn_fenced_abort:600: With error INVALID_PRODUCER_EPOCH)" failed at 0105-transactions_mock.c:667:do_test_txn_fenced_abort() at Thu Oct 13 12:04:14 2022: ###
Test assertion failed: "errors_cnt == 0": Expected error count 0 for API EndTxn, found 1`